code/cfile/cfile.cpp

```diff
@@ -593,6 +593,24 @@ int cf_rename(const char *old_name, const char *name, int dir_type)
 
 }
 
+static void mkdir_recursive(const char *dir) {
+    char tmp[256];
```

Comment:
This should use an SCP_string to make sure too long path elements don't cause a buffer overflow.